executor/simple.go

```diff
@@ -753,9 +753,9 @@ func (e *SimpleExec) executeCreateUser(ctx context.Context, s *ast.CreateUserStm
 
 	sql := new(strings.Builder)
 	if s.IsCreateRole {
-		sqlexec.MustFormatSQL(sql, `INSERT INTO %n.%n (Host, User, authentication_string, Account_locked) VALUES `, mysql.SystemDB, mysql.UserTable)
+		sqlexec.MustFormatSQL(sql, `INSERT INTO %n.%n (Host, User, authentication_string, plugin, Account_locked) VALUES `, mysql.SystemDB, mysql.UserTable)
 	} else {
-		sqlexec.MustFormatSQL(sql, `INSERT INTO %n.%n (Host, User, authentication_string) VALUES `, mysql.SystemDB, mysql.UserTable)
+		sqlexec.MustFormatSQL(sql, `INSERT INTO %n.%n (Host, User, authentication_string, plugin) VALUES `, mysql.SystemDB, mysql.UserTable)
```

Comment:
Can you add some tests - maybe here you can test the `CREATE USER` statement and then verify in `SHOW CREATE USER` that the plugin is correct? (for both mysql_native_password and sha2 caching). This will prove end-to-end the plugin meta data is correct.

See other privilege integration tests in `privilege/privileges/privileges_test.go`.

Reply:
@lysu Do you have any suggestion for the tests?

Reply:
I think [TestShowCreateUser](https://github.com/pingcap/tidb/blob/b64f9007cfec31c4eb2455c1b89ae441a6c6b79c/executor/show_test.go#L397-L436) already does most of this. 

Creating users with `caching_sha2_password` isn't yet supported with this:
```
mysql> CREATE USER 'fooobaaar'@'%' IDENTIFIED WITH 'caching_sha2_password' AS '$A$005$@<h4D7f<Cv &dnWClPZQUoJVM0tlL8EK3xrH6Plxdiozhfk/bAKhaisVG9';
ERROR 1827 (HY000): The password hash doesn't have the expected format. Check if the correct password algorithm is being used with the PASSWORD() function.
```
I'll add a test which creates an entry in mysql.user with `caching_sha2_password` and then checks the show output.